### PR TITLE
sql/sqlite: bug fixes and migration support

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

File: internal/integration/sqlite_test.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+func (t *liteTest) dropTables(names ...string) {
+	t.Cleanup(func() {
+		for i := range names {
+			_, err := t.db.Exec("DROP TABLE IF EXISTS " + names[i])
+			require.NoError(t.T, err, "drop tables %q", names[i])
+		}
+	})
+}
```
Comment (yonidavidson) [inline internal/integration/sqlite_test.go:164]: Are you adding TestSqlite_Ent later?

File: sql/sqlite/migrate.go [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+	if alterable(modify) {
+		return m.alterTable(ctx, modify)
+	}
+	if err := m.skipConstraints(ctx, func(ctx context.Context) error {
+		newT := *modify.T
+		indexes := newT.Indexes
+		newT.Indexes = nil
+		newT.Name = "new_" + newT.Name
+		// Create a new table with a temporary name, and copy the existing rows to it.
+		if err := m.addTable(ctx, &schema.AddTable{T: &newT}); err != nil {
```
Comment (yonidavidson) [inline sql/sqlite/migrate.go:94]: Wow!

File: internal/integration/sqlite_test.go (hunk 1 above)
Comment (a8m) [reply internal/integration/sqlite_test.go:164]: Yes. It’s WIP 🙃

**Comment by a8m** [comment]
> Looks great, just missing the ent integration tests, is that for another PR?

Yes.

**Pull request merged**